MediaBrowser.MediaEncoding/Attachments/AttachmentExtractor.cs

```diff
@@ -230,13 +231,166 @@ public async Task<(MediaAttachment Attachment, Stream Stream)> GetAttachment(Bas
+
             var outputPath = GetAttachmentCachePath(mediaPath, mediaSource, mediaAttachment.Index);
             await ExtractAttachment(inputFile, mediaSource, mediaAttachment.Index, outputPath, cancellationToken)
                 .ConfigureAwait(false);
 
             return outputPath;
         }
 
+        private async Task CacheAllAttachments(
+            string mediaPath,
+            string inputFile,
+            MediaSourceInfo mediaSource,
+            CancellationToken cancellationToken)
+        {
+            var locks = new List<AsyncKeyedLockReleaser<string>>();
+            var extractableAttachmentIds = new List<int>();
+
+            try
+            {
+                foreach (var attachment in mediaSource.MediaAttachments)
+                {
+                    var outputPath = GetAttachmentCachePath(mediaPath, mediaSource, attachment.Index);
+
+                    var @lock = _semaphoreLocks.GetOrAdd(outputPath);
+                    await @lock.SemaphoreSlim.WaitAsync(cancellationToken).ConfigureAwait(false);
+
+                    if (File.Exists(outputPath))
+                    {
+                        @lock.Dispose();
+                        continue;
+                    }
+
+                    locks.Add(@lock);
+                    extractableAttachmentIds.Add(attachment.Index);
+                }
+
+                if (extractableAttachmentIds.Count > 0)
+                {
+                    await CacheAllAttachmentsInternal(mediaPath, inputFile, mediaSource, extractableAttachmentIds, cancellationToken).ConfigureAwait(false);
+                }
+            }
+            catch (Exception ex)
+            {
+                _logger.LogWarning(ex, "Unable to cache media attachments for File:{File}", mediaPath);
+            }
+            finally
+            {
+                foreach (var @lock in locks)
+                {
+                    @lock.Dispose();
+                }
+            }
+        }
+
+        private async Task CacheAllAttachmentsInternal(
+            string mediaPath,
+            string inputFile,
+            MediaSourceInfo mediaSource,
+            List<int> extractableAttachmentIds,
+            CancellationToken cancellationToken)
+        {
+            var outputPaths = new List<string>();
+            var processArgs = string.Empty;
+
+            foreach (var attachmentId in extractableAttachmentIds)
+            {
+                var outputPath = GetAttachmentCachePath(mediaPath, mediaSource, attachmentId);
+
+                Directory.CreateDirectory(Path.GetDirectoryName(outputPath) ?? throw new FileNotFoundException($"Calculated path ({outputPath}) is not valid."));
+
+                outputPaths.Add(outputPath);
+                processArgs += string.Format(
+                    CultureInfo.InvariantCulture,
+                    " -dump_attachment:{0} \"{1}\"",
```

Comment:
Maybe add `:t`, like `-dump_attachment:t:{0}`? As per https://ffmpeg.org/ffmpeg-all.html#toc-Main-options

> stream_type[:additional_stream_specifier]
> stream_type is one of following: ’v’ or ’V’ for video, ’a’ for audio, ’s’ for subtitle, ’d’ for data, and ’t’ for attachments. ’v’ matches all video streams, ’V’ only matches video streams which are not attached pictures, video thumbnails or cover arts. If additional_stream_specifier is used, then it matches streams which both have this type and match the additional_stream_specifier. Otherwise, it matches all streams of the specified type.